impl/maven-impl/src/main/java/org/apache/maven/impl/model/rootlocator/DefaultRootLocator.java

```diff
@@ -64,8 +66,14 @@ public Path findMandatoryRoot(@Nonnull Path basedir) {
             rootDirectory = rdf.orElseThrow(() -> new IllegalStateException(getNoRootMessage()));
             logger.warn(getNoRootMessage());
         } else {
-            if (rdf.isPresent() && !Objects.equals(rootDirectory, rdf.get())) {
-                logger.warn("Project root directory and multiModuleProjectDirectory are not aligned");
+            if (rdf.isPresent()) {
+                try {
+                    if (!Files.isSameFile(rootDirectory, rdf.orElseThrow())) {
+                        logger.warn("Project root directory and multiModuleProjectDirectory are not aligned");
+                    }
+                } catch (IOException e) {
+                    throw new UncheckedIOException("findMandatoryRoot failed", e);
```

Comment:
UncheckedIOException is the wrong exception type here. Possibly this should return null or throw some Maven specific checked exception. Previously it threw IllegalStateException which is not really right either, but might be OK if this really "can't" happen.